app/javascript/flavours/glitch/components/status_reactions.js

```diff
@@ -0,0 +1,176 @@
+import ImmutablePureComponent from 'react-immutable-pure-component';
+import PropTypes from 'prop-types';
+import ImmutablePropTypes from 'react-immutable-proptypes';
+import { maxReactions, reduceMotion } from '../initial_state';
+import spring from 'react-motion/lib/spring';
+import TransitionMotion from 'react-motion/lib/TransitionMotion';
+import classNames from 'classnames';
+import React from 'react';
+import unicodeMapping from '../features/emoji/emoji_unicode_mapping_light';
+import AnimatedNumber from './animated_number';
+import { assetHost } from '../utils/config';
+import { autoPlayGif } from '../initial_state';
+
+export default class StatusReactions extends ImmutablePureComponent {
+
+  static propTypes = {
+    statusId: PropTypes.string.isRequired,
+    reactions: ImmutablePropTypes.list.isRequired,
+    addReaction: PropTypes.func.isRequired,
+    removeReaction: PropTypes.func.isRequired,
+    emojiMap: ImmutablePropTypes.map.isRequired,
+  };
+
+  handleEmojiPick = data => {
+    const { addReaction, statusId } = this.props;
+    addReaction(statusId, data.native.replace(/:/g, ''));
+  }
+
+  willEnter() {
+    return { scale: reduceMotion ? 1 : 0 };
+  }
+
+  willLeave() {
+    return { scale: reduceMotion ? 0 : spring(0, { stiffness: 170, damping: 26 }) };
+  }
+
+  render() {
+    const { reactions } = this.props;
+    const visibleReactions = reactions
+      .filter(x => x.get('count') > 0)
+      .sort((a, b) => b.get('count') - a.get('count'))
+      .filter((_, i) => i < maxReactions);
```

Comment:
The limit for how many reactions are displayed should probably different from the amount of reaction one user can set.
Let's say an admin limits reaction per user to 1, now only the most used emoji is displayed on a post.
I propose a new variable like: `MAX_VISIBLE_REACTIONS`
And change the default of max visible to 6, which would be 1 row. Looks cleaner imo.

Reply:
Good point, I'll throw that into the app settings modal.  And now that I think about it, it might also be useful to have a button that quickly expands all reactions for a single toot.  What do you think?

Reply:
I'm not sure about a button, since I think it would look weird.
Maybe a column setting or user preference to show all reactions and/or show all in the detailed status view?

Reply:
Quick mock-up for a button (missing proper styling, but good enough to visualise):
![Screenshot_2022-11-30_13-28-22](https://user-images.githubusercontent.com/29483052/204796609-2451e1cc-d34b-4454-96eb-5c3156a06239.png)
I used the plus icon because it was the quickest I could find which fits. 
Might be confused with "add reaction" though

Reply:
I think using a plus button would be kind of confusing because that's already used for adding reactions.  Also, announcement reactions have their plus button right next to the existing emojis, just where that expand button would be.  The way I'm currently doing it is only limiting in the timeline view, and always display all reactions when you click on the status to open the detailed view.

Reply:
Agree